File: StoneGame.cpp
class Solution {
public:
    bool stoneGame(vector<int>& piles) {
        //min. runtime(0 ms) S(C(N)=(O(1))) and (T(C(N)=O(n))due to Level wise iteration and requires non additonal space
        int n;int sum1=0;//variables declare and sum initization
        for(int i=0;i<piles.size();i++){//iterating to Pile's length
            if(piles[i]==0)return 0;//piles initizlation 
            else if(n%2==0 && n>0)return false;//prinitng false to   at  even stones piles positive number non winning condition 
            else if(piles[i]>0) piles.begin()=piles.end();return true;//printing true to the winner from piles begin till end at its most number
        }return false;}};//printing false for nonwining cond
    

File: StoneGAME.py
class Solution:
    def stoneGame(self, piles: List[int]) -> bool:
        #T(C(N)=O(N)) and S(C(N))=O(1) as it requires non memory space allocation iteratively 
        n=int()#Stone(n) declare
        for i in range(len(piles)):#iterating Through Pile's Stack's Length
            if piles[i]==0:return None#Initilaizing Pile
            elif n%2==0 and n>0:return False#Printing False to the Pile's Even Number of Stones
            elif piles[i]>0:piles[i]=piles[i-1];return True#printing True to Winner with the Pile's Start till its End  Odd Number Stone 
        return False#printing False to the Unwinner 
    #def stoneGame(self, piles: List[int]) -> bool:return True
